front-end/js/jump.js

```diff
@@ -330,6 +330,371 @@ function startJump(viewId, d, jump, optionalArgs) {
+
+        updateAttrs.selectAll(".attr-inputs").on("click", function(d) {
+            d3.event.stopPropagation();
+        });
+    }
+
+    // add listener to save changes button, sends updates to backend
+    for (let i = 0; i < updateBtnIds.length; i++) {
+        let btnObj = d3.select(`#${updateBtnIds[i]}`);
+        let updateAttr = btnObj.attr("updateAttr");
+        let inputId = "#attr-input-" + i;
+        btnObj.on("click", function(d) {
+            let attrValue = d3.select(inputId).property("value");
+            const viewClass = ".view_" + viewId;
+            let newAttrValues = [];
+            let objectKV = {};
+            for (let i = 0; i < directMappedColNames.length; i++) {
+                let colName = directMappedColNames[i];
+                newAttrValues.push(p[colName]);
+            }
+            for (let i = 0; i < directMappedColNames.length; i++) {
+                let colName = directMappedColNames[i];
+                objectKV[colName] = newAttrValues[i];
+            }
+            objectKV[updateAttr] = attrValue;
+
+            // reverse function code
+            const reverseFuncString =
+                gvd.curCanvas.layers[layerId].transform.reverseFunctions[
+                    updateAttr
+                ];
+
+            const reverseFunc = Function(reverseFuncString)();
+            let width = gvd.curCanvas.w;
+            let height = gvd.curCanvas.h;
+            objectKV = reverseFunc(objectKV, width, height);
+
+            const projName = globalVar.project.name;
+            doDBUpdate(
+                viewId,
+                canvasId,
+                layerId,
+                tableName,
+                objectKV,
+                projName
+            );
+            // re-load dynamic data from db
+            getCurCanvas(viewId);
+            if (!gvd.animation) {
+                var curViewport = d3
+                    .select(viewClass + ".mainsvg:not(.static)")
+                    .attr("viewBox")
+                    .split(" ");
+                RefreshDynamicLayers(viewId, curViewport[0], curViewport[1]);
+            }
+            removePopovers(viewId);
+        });
+    }
+
+    // finally position updates popover according to event x/y and its width/height
+    // position jump popover according to event x/y and its width/height
+    var popoverHeight = d3
+        .select(viewClass + ".jumppopover")
+        .node()
+        .getBoundingClientRect().height;
+    d3.select(viewClass + ".jumppopover")
+        .style("left", d3.event.pageX + "px")
+        .style("top", d3.event.pageY - popoverHeight / 2 + "px");
+}
+
+function registerDragNDropUpdateListener(viewId, p, layerId) {
+    var gvd = globalVar.views[viewId];
+    var viewClass = ".view_" + viewId;
+    let layerObj = gvd.curCanvas.layers[layerId];
+    var allowsXUpdate = false;
+    var allowsYUpdate = false;
+    if ("x" in layerObj.transform.reverseFunctions) allowsXUpdate = true;
```

Comment:
Let me review this code again, I'm trying to think if there's a good reason why I can't use `cx` directly and not assume `x` is an attribute. You're right that this would be better if possible.